From 9607e9858d681f5ad2e1677f5dbbb1fb61bd8846 Mon Sep 17 00:00:00 2001
From: Jose Quaresma <quaresma.jose@gmail.com>
Date: Sun, 18 Oct 2020 10:40:59 +0100
Subject: [PATCH 4/4] Currently gstreamer configuration depends on whether
 setcap is found on the host system. Turn this into a configure option to make
 builds deterinistic.

RP 2020/2/19
Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>
Upstream-Status: Pending
---
 libs/gst/helpers/meson.build | 7 ++++++-
 meson_options.txt            | 1 +
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/libs/gst/helpers/meson.build b/libs/gst/helpers/meson.build
index 9e5a4ae..c2e540f 100644
--- a/libs/gst/helpers/meson.build
+++ b/libs/gst/helpers/meson.build
@@ -66,7 +66,12 @@ if have_ptp
       description : 'getifaddrs() and AF_LINK is available')
   endif
 
-  setcap_prog = find_program('setcap', '/usr/sbin/setcap', '/sbin/setcap', required : false)
+  setcap_feature = get_option('setcap')
+  if setcap_feature.disabled()
+    setcap_prog = find_program('dontexist', required : false)
+  else
+    setcap_prog = find_program('setcap', '/usr/sbin/setcap', '/sbin/setcap', required : false)
+  endif
   cap_dep = dependency('libcap', required: false)
 
   # user/group to change to in gst-ptp-helper
diff --git a/meson_options.txt b/meson_options.txt
index 346c423..03338ad 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -25,6 +25,7 @@ option('libdw', type : 'feature', value : 'auto', description : 'Use libdw to ge
 option('dbghelp', type : 'feature', value : 'auto', description : 'Use dbghelp to generate backtraces')
 option('bash-completion', type : 'feature', value : 'auto', description : 'Install bash completion files')
 option('coretracers', type : 'feature', value : 'auto', description : 'Build coretracers plugin')
+option('setcap', type : 'feature', value : 'auto', description : 'Use setcap')
 
 # Common feature options
 option('examples', type : 'feature', value : 'auto', yield : true)
-- 
2.28.0

